iothub/device/src/DeviceAuthenticationWithSakRefresh.cs

```diff
@@ -11,18 +11,18 @@ namespace Microsoft.Azure.Devices.Client
     // Implementing SAS Token refresh based on a SharedAccessKey (SAK).
     internal class DeviceAuthenticationWithSakRefresh : DeviceAuthenticationWithTokenRefresh
     {
-        private IotHubConnectionString _connectionString;
+        private readonly IotHubConnectionString _connectionString;
 
         public DeviceAuthenticationWithSakRefresh(
-            string deviceId, 
+            string deviceId,
             IotHubConnectionString connectionString) : base(deviceId)
         {
-            _connectionString = connectionString;
+            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
```

Comment:
why do we not want to throw this exception?

Reply:
I added the throw. What do you mean?

Reply:
I don't understand the question @abhipsaMisra :D